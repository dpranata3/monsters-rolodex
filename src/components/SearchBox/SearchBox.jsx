import React from "react";

const searchBox = ({ onChangeSearch }) => {
  return (
    <input
      className="search-box"
      type="text"
      placeholder="search monsters"
      onChange={(event) => onChangeSearch(event.target.value)}
    />
  );
};

export default searchBox;
