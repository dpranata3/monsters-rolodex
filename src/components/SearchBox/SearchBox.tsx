import { ChangeEvent } from "react";
import styled from "styled-components";
interface ISearchBoxProps {
  placeholder?: string;
  onSearchChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

const searchBox = ({ onSearchChange, placeholder }: ISearchBoxProps) => {
  return (
    <Container
      id="search-box"
      className="search-box"
      type="text"
      placeholder={placeholder}
      onChange={onSearchChange}
    />
  );
};

export default searchBox;

const Container = styled.input`
  border-radius: 4;
  border: none;
  height: 44px;
  line-height: 30px;
  margin: auto;
  outline: none;
  padding: 10px;
  width: 150px;
`;