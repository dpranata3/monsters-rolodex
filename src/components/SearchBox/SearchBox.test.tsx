import { shallow } from "enzyme";

import SearchBox from "./SearchBox";

describe("SearchBox test suite", () => {
  const mockSearchFn = jest.fn();
  const testSearchState = "test input"

  it("expect to render SearchBox component", () => {
    expect(
      shallow(<SearchBox onSearchChange={mockSearchFn} />)
    ).toMatchSnapshot();
  });

  it("expect search to receive input", () => {
    const wrapper = shallow(<SearchBox onSearchChange={mockSearchFn}/>);

    wrapper
      .find('[id="search-box"]')
      .simulate("change", { target: { value: "test input" } });

    expect(testSearchState).toEqual("test input");
  })
});


