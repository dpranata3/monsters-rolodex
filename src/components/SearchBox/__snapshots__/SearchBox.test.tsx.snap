// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox test suite expect to render SearchBox component 1`] = `ShallowWrapper {}`;
